hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/validators/TokenUpdateValidator.java

```diff
@@ -62,6 +62,10 @@ public ValidationResult validateSemantics(
         if (op.hasMemo()) {
             context.attributeValidator().validateMemo(op.memo());
         }
+        // validate metadata
+        if (op.metadata() != null && !op.metadata().toString().isEmpty()) {
+            validator.validateTokenMetadata(String.valueOf(op.metadata()), tokensConfig);
+        }
```

Comment:
Is the `toString()` of metadata dependable, and should metadata always be interpretable as text? Based on the protobuf definition (`Metadata of the created token definition`), I would expect this to be something more like
```
// use `Arrays.isNullOrEmpty(byte[])` to check for an effectively empty byte array, 
// which - unlike toString() – won't use a potentially lossy transform of the original bytes 

if (op.metadata() != null && Arrays.isNullOrEmpty(op.getMetadata().toByteArray()) {
    // no potentially lossy conversion to a string
    validator.validateTokenMetadata(op.metadata(), tokensConfig);
}
```

Reply:
Good suggestion @mhess-swl ! 💯
Will modify accordingly